src/window-rules.c

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#define _POSIX_C_SOURCE 200809L
+#include "config.h"
+#include <assert.h>
+#include <cairo.h>
+#include <glib.h>
+#include <strings.h>
+#include <wlr/util/log.h>
+#include "action.h"
+#include "common/match.h"
+#include "config/rcxml.h"
+#include "labwc.h"
+#include "view.h"
+#include "window-rules.h"
+
+static const char *
+get_identifier(struct view *view)
+{
+	switch (view->type) {
+	case LAB_XDG_SHELL_VIEW:
+		return view_get_string_prop(view, "app_id");
+#if HAVE_XWAYLAND
+	case LAB_XWAYLAND_VIEW:
+		return view_get_string_prop(view, "class");
+#endif
+	}
+	return "";
```

Comment:
`src/xwayland.c` responds to a `app_id` request as well so I assume we can just always ask for `app_id`.

Reply:
Yes, although it's clearer what's going on the way it is written, don't you think?

Reply:
We'd get rid of the `#if HAVE_XWAYLAND` though. Maybe we could add a handler for the string `identifier` instead? And then have either `src/view.c` or `src/xdg.c` / `src/xwayland.c` handle it appropriately.

Reply:
Okay. Done. Have got for a simple call for `prop("app_id")`.